src/librustc_trans/debuginfo/metadata.rs

```diff
@@ -1517,22 +1520,66 @@ enum EnumDiscriminantInfo {
     NoDiscriminant
 }
 
+//There's no good place to put this.
+//It is used here and in prepare_enum_metadata only.
+fn discriminant_to_ty<'a, 'tcx>(tcx: &ty::TyCtxt<'a, 'tcx, 'tcx>,
```

Comment:
I think it is more confusing in `debuginfo`. Converting a layout integer to a Rust `Ty` sounds quite natural.